src/components/lyricseditor/lyricseditor.js

```diff
@@ -0,0 +1,393 @@
+        showLyricsPreview(lyrics.innerHTML);
+    }
+
+    const btnDownload = dom.parentWithClass(e.target, 'btnDownload');
+    if (btnDownload) {
+        lyricsId = btnDownload.getAttribute('data-lyricsid');
+        context = dom.parentWithClass(btnDownload, 'lyricsEditorDialog');
+        downloadRemoteLyrics(context, lyricsId);
+    }
+}
+
+function showLyricsPreview(lyrics) {
+    const dialogOptions = {
+        removeOnClose: true,
+        scrollY: false
+    };
+
+    if (layoutManager.tv) {
+        dialogOptions.size = 'fullscreen';
+    } else {
+        dialogOptions.size = 'small';
+    }
+
+    const dlg = dialogHelper.createDialog(dialogOptions);
+
+    dlg.classList.add('formDialog');
+    dlg.classList.add('lyricsEditorDialog');
+
+    dlg.innerHTML = globalize.translateHtml(templatePreview, 'core');
+
+    dlg.querySelector('.lyricsPreview').innerHTML = lyrics;
+
+    dlg.querySelector('.btnCancel').addEventListener('click', function () {
+        dialogHelper.close(dlg);
+    });
+
+    dialogHelper.open(dlg);
+}
+
+function showOptions(button, context, lyricsId, lyrics) {
+    const items = [];
+
+    items.push({
+        name: globalize.translate('PreviewLyrics'),
+        id: 'preview'
+    }
+    , {
+        name: globalize.translate('Download'),
+        id: 'download'
+    });
+
+    import('../actionSheet/actionSheet').then((actionsheet) => {
+        actionsheet.show({
+            items: items,
+            positionTo: button
+
+        }).then(function (id) {
+            if (id === 'download') {
+                downloadRemoteLyrics(context, lyricsId);
+            }
+            if (id === 'preview') {
+                showLyricsPreview(lyrics);
+            }
+        });
+    });
+}
+
+function centerFocus(elem, horiz, on) {
+    import('../../scripts/scrollHelper').then(({ default: scrollHelper }) => {
+        const fn = on ? 'on' : 'off';
+        scrollHelper.centerFocus[fn](elem, horiz);
+    });
+}
+
+function onOpenUploadMenu(e) {
+    const dialog = dom.parentWithClass(e.target, 'lyricsEditorDialog');
+    const apiClient = ServerConnections.getApiClient(currentItem.ServerId);
+
+    import('../lyricsuploader/lyricsuploader').then(({ default: lyricsUploader }) => {
+        lyricsUploader.show({
+            itemId: currentItem.Id,
+            serverId: currentItem.ServerId
+        }).then(function (hasChanged) {
+            if (hasChanged) {
+                hasChanges = true;
+                reload(dialog, apiClient, currentItem.Id);
+            }
+        });
+    });
+}
+
+function onDeleteLyrics(e) {
+    deleteLyrics(currentItem).then(() => {
+        hasChanges = true;
+        const context = dom.parentWithClass(e.target, 'formDialogContent');
+        const apiClient = ServerConnections.getApiClient(currentItem.ServerId);
+        reload(context, apiClient, currentItem.Id);
+    }).catch(() => {
+        console.warn('Failed to delete lyrics for', currentItem.Name);
+    });
```

Comment:
I should have been more clear here. The `deleteLyrics` modal already handles if an error occurs while deleting, but our modal code uses kind of a bad pattern where closing a modal rejects, so really we just need to ignore it.

```suggestion
    }).catch(() => {
        // delete dialog closed
    });
```

Reply:
Yeah, that's what I assumed. But doing nothing in a catch statement doesn't really feel right to me, so I added some console logs there.😆